engine/render/src/render/render_script.cpp

```diff
@@ -1372,29 +1409,63 @@ namespace dmRender
 
-            dmGraphics::HTexture texture = dmGraphics::GetRenderTargetTexture(render_target, buffer_type);
+            if (!dmGraphics::IsAssetHandleValid(i->m_RenderContext->m_GraphicsContext, asset_handle))
+            {
+                char buf[128];
+                return luaL_error(L, "Texture handle '%s' is not valid.", AssetHandleToString(asset_handle, buf, sizeof(buf)));
+            }
+
+            if (asset_type == dmGraphics::ASSET_TYPE_RENDER_TARGET)
+            {
+                dmGraphics::BufferType buffer_type = dmGraphics::BUFFER_TYPE_COLOR0_BIT;
+                if (lua_isnumber(L, 3))
+                {
+                    buffer_type = (dmGraphics::BufferType) lua_tonumber(L, 3);
+                }
+
+                texture = dmGraphics::GetRenderTargetTexture(asset_handle, buffer_type);
+
+                if (texture == 0)
+                {
+                    char buf[128];
+                    return luaL_error(L, "Render target '%s' does not have a texture for the specified buffer type.", AssetHandleToString(asset_handle, buf, sizeof(buf)));
+                }
+            }
+            else if (asset_type == dmGraphics::ASSET_TYPE_TEXTURE)
+            {
+                texture = (dmGraphics::HTexture) asset_handle;
+            }
 
+            // Texture can potentially still be zero if it's an attachment texture
             if(texture != 0)
             {
-                if (InsertCommand(i, Command(COMMAND_TYPE_ENABLE_TEXTURE, unit, (uintptr_t)texture)))
+                // JG: We are passing in the opaque handle here and not the asset handle, this is a workaround for 32 bit systems
+                //     because we are using uintptr_t that is defined as a uint32_t on those platforms. We reconstruct the actual handle
+                //     before actual passing in render_command.cpp that processes the command. And alternative would be to use 64b values for the operands?
+                if (InsertCommand(i, Command(COMMAND_TYPE_ENABLE_TEXTURE, unit, (uintptr_t) dmGraphics::GetOpaqueHandle(texture))))
+                {
                     return 0;
+                }
                 else
+                {
                     return luaL_error(L, "Command buffer is full (%d).", i->m_CommandBuffer.Capacity());
+                }
             }
-            return luaL_error(L, "Render target does not have a texture for the specified buffer type.");
+
+            char buf[128];
+            return luaL_error(L, "Texture handle '%s' is not valid.", AssetHandleToString(asset_handle, buf, sizeof(buf)));
         }
         else
         {
-            return luaL_error(L, "%s.enable_texture(unit, render_target, buffer_type) called with illegal parameters.", RENDER_SCRIPT_LIB_NAME);
+            return luaL_error(L, "%s.enable_texture(unit, render_target|handle, buffer_type) for unit %d called with illegal parameters.", RENDER_SCRIPT_LIB_NAME, unit);
```

Comment:
I guess the render target should be represented as a handle now?
I.e. should it only say "%s.enable_texture(unit, handle, buffer_type) ..." ?

Or am I missing something?

Reply:
Aah yes, good catch